client/elasticClientCommon.go

```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
 		return fmt.Errorf("%w cannot read elastic response body bytes", err)
 	}
 
-	bulkResponse := &BulkRequestResponse{}
-	err = json.Unmarshal(bodyBytes, bulkResponse)
+	return extractErrorFromBulkBodyResponseBytes(bodyBytes, index)
+}
+
+func extractErrorFromBulkBodyResponseBytes(bodyBytes []byte, index string) error {
+	if index == indexer.TransactionsIndex {
+		return searchForErrorsUpdate(bodyBytes)
+	}
+
+	return searchForErrorsIndex(bodyBytes)
+}
+
+func searchForErrorsUpdate(bodyBytes []byte) error {
+	response := BulkRequestResponseUpdate{}
```

Comment:
a lot of duplicated code between searchForErrorsUpdate and searchForErrorsIndex

Reply:
refactored.